src/api/miro.api.ts

```diff
@@ -1,61 +1,158 @@
-import { CARD_TITLE_TRKR_ITEMID_FILTER_REGEX } from '../constants/regular-expressions';
+import { AppCard } from "@mirohq/websdk-types";
+import { EnhancedStore } from "@reduxjs/toolkit";
+import { CodeBeamerItem } from "../models/codebeamer-item.if";
+import { store } from "../store/store";
+import addCardFields from "./utils/addCardFields";
+import getCardTitle from "./utils/getCardTitle";
+import getItemColorField from "./utils/getItemColorField";
+
+import { DescriptionFormat } from "../enums/descriptionFormat.enum";
+import getConcentricCircleCoords from "./utils/getConcentricCircleCoords";
+import { CardSpawningMethod } from "../enums/cardSpawningMethod.enum";
+import getRandomCoordSetPerSubject from "./utils/getRandomCoordSetPerSubject";
+import getSnailCoordSetPerSubject from "./utils/getSnailCoords";
+import { CARD_TITLE_TRKR_ITEMID_FILTER_REGEX } from "../constants/regular-expressions";
+import getAppCardIds from "./utils/getAppCardIds";
+import doesConnectorExist from "./utils/doesConnectorExist";
+import { Association } from "../models/api-query-types";
+import { RelationshipType } from "../enums/associationRelationshipType.enum";
+import { getColorForRelationshipType } from "./utils/getColorForRelationshipType";
 
 /**
  * Create a new app card base on a codeBeamer item
  * @param item The item to base the card on
  * @param distributedSpawning Set to false for the item to be spawned roughly in the viewport's center, to true for distribution using the full vp dimensions.
  */
 export async function createAppCard(
-	item: CodeBeamerItem,
-	cardSpawningMethod: CardSpawningMethod = CardSpawningMethod.CONCENTRIC_CIRCLES
+  item: CodeBeamerItem,
+  cardSpawningMethod: CardSpawningMethod = CardSpawningMethod.CONCENTRIC_CIRCLES
+) {
+  const card: Partial<AppCard> = await convertToCardData(item);
+  let coords: { x: number; y: number };
+
+  switch (cardSpawningMethod) {
+    case CardSpawningMethod.CONCENTRIC_CIRCLES: {
+      coords = await getConcentricCircleCoords(item);
+      break;
+    }
+    case CardSpawningMethod.RANDOM_IN_VIEWPORT: {
+      coords = await getRandomCoordSetPerSubject(item);
+      break;
+    }
+    case CardSpawningMethod.SNAIL: {
+      coords = await getSnailCoordSetPerSubject(item);
+      break;
+    }
+  }
+
+  card.x = coords.x;
+  card.y = coords.y;
+
+  try {
+    const widget = await miro.board.createAppCard({
+      ...card,
+    });
+    await widget.setMetadata("item", {
+      id: item.id,
+      tracker: {
+        id: item.tracker.id,
+      },
+    });
+  } catch (error) {
+    console.error(error);
+  }
+}
+
+//in a for loop get detailed data for each association and then call createConnector() to create each connector
+export async function createConnectorsForDownstreamRefsAndAssociation(
+  startCardId: string,
+  downstreamRefs: [number],
+  associations: [{ associationId: number; targetItemId: number }],
+  boardData: BoardNode[]
```

Comment:
> Lastly, one major issue that seems to appear when using the feature a lot is this console log, and the complete shutdown of any miro-sdk functionality along with it:
> 
> ![image](https://user-images.githubusercontent.com/70019423/226275033-db90280c-d180-40de-8991-17c76177c0d8.png)
> 

You wrote

> I implemented this [...] it works I get the data from the board at the start and then use that to do all the rest of the logic, seems kind of laggy when spamming show/hide

Which doesn't sound entirely conclusive. So let's open this actionable thread for this.

Besides the laggyness experienced for an item with more relations, there also seems to be an inconsistency in how the button reacts once relations have just been imported:

![cbc_assoc_button](https://user-images.githubusercontent.com/70019423/227848117-c821b36f-1036-4eae-926f-452c98c79f8e.gif)

Reply:
Additionally, did you evaluate whether the refactoring solves the root problem? If so, to what degree? Eventually, the limit comes into play again of course, but can it be reached in considerable time?

Reply:
It would truly appear that the problem has been solved.

I've been toggling between showing and hiding 81 dependencies at a time for 10 times and not run into any inconsistency or limit.

![image](https://user-images.githubusercontent.com/70019423/236993649-8508e887-571a-4f6f-a329-86d7d2423ce5.png)

Can you please summarize what the problem was and how we solved it, maybe in the (or a new) development / contribution documentation, so we have a reference about it in future, @C2H6Ethan ?